tests/e2e/common.go

```diff
@@ -294,3 +284,23 @@ func watchPods() {
 		}
 	}
 }
+
+// helper func to write `kubectl get -o yaml` output to file
+func writeToFile(fileName string, resource Output) error {
+	file, err := os.OpenFile(fileName, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	if err != nil {
+		fmt.Printf("Failed to open log file: %v\n", err)
+		return err
+	}
+	defer file.Close()
```

Comment:
this is interesting that the file is being closed with every write. Might be a little less performant to continually open and close, but I guess it's working. The alternative would be to maintain a list of all open files and then close them all at the end. But since it's working, I'm fine as is. Maybe we can just add a "TODO" comment: "for better performance, don't close file after every write"

Reply:
That's right, this approach is much more cautious just cause I don't want any files to continue to be open for too long or not close to cause an issue.

Keeping a list of every open file would be good, we could automatically close everything when a test fails as well too.

Reply:
I'm fine if you just want to add a "TODO" for now - up to you

Reply:
I think I could add this change to this PR since we're adding the `writeToFile` func here and the other comment to consolidate further could be its own PR.